sdk/python/jobs/parallel/1a_oj_sales_prediction/oj_sales_prediction.ipynb

```diff
@@ -0,0 +1,386 @@
+    "        entry_script=\"parallel_train.py\",\n",
+    "        environment=Environment(\n",
+    "            image=\"mcr.microsoft.com/azureml/openmpi3.1.2-ubuntu18.04\",\n",
+    "            conda_file=\"./src/parallel_train/conda.yml\",\n",
+    "        ),\n",
+    "        program_arguments=\"--drop_cols ${{inputs.drop_cols}} \"  # Passthrough input parameters into parallel_train script.\n",
+    "        \"--target_col ${{inputs.target_col}} \"\n",
+    "        \"--date_col ${{inputs.date_col}} \"\n",
+    "        \"--lagging_orders ${{inputs.lagging_orders}} \"\n",
+    "        \"--model_folder ${{outputs.model_folder}} \",\n",
+    "        append_row_to=\"${{outputs.model_perf}}\",  # Define where to output the aggregated returns from each mini-batches.\n",
+    "    ),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 3. Build pipeline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1673860087207
+    },
+    "name": "build-pipeline"
+   },
+   "outputs": [],
+   "source": [
+    "# Declare the overall input of the job.\n",
+    "input_oj_data = Input(\n",
+    "    path=\"./oj_sales_data/oj_sales_data.csv\",\n",
+    "    type=AssetTypes.URI_FILE,\n",
+    "    mode=InputOutputModes.RO_MOUNT,\n",
+    ")\n",
+    "\n",
+    "# Declare pipeline structure.\n",
+    "@pipeline(\n",
+    "    display_name=\"parallel job for oj many model training\",\n",
+    ")\n",
+    "def partition_job_in_pipeline(\n",
+    "    pipeline_input_data,\n",
+    "):\n",
+    "    # Declare 1st data partition command job.\n",
+    "    partition_job = partition_data(\n",
+    "        data_source=pipeline_input_data,\n",
+    "        partition_keys=\"Store,Brand\",\n",
+    "    )\n",
+    "\n",
+    "    # Declare 2nd parallel model training job.\n",
+    "    parallel_train = many_model_training_with_partition_keys(\n",
+    "        data_source=partition_job.outputs.tabular_output_data,\n",
+    "        drop_cols=\"Revenue,Advert,Store,Brand\",\n",
+    "        target_col=\"Quantity\",\n",
+    "        date_col=\"WeekStarting\",\n",
+    "        lagging_orders=\"1,2,3,4,5,6\",\n",
+    "    )\n",
+    "\n",
+    "    # User could override parallel job run-level property when invoke that parallel job/component in pipeline.\n",
+    "    parallel_train.resources.instance_count = 3\n",
+    "    parallel_train.max_concurrency_per_instance = 2\n",
+    "    parallel_train.mini_batch_error_threshold = 10\n",
+    "    parallel_train.outputs.model_perf.path = (\n",
+    "        f\"azureml://datastores/workspaceblobstore/paths/${{name}}/my_append_output.csv\"\n",
+    "    )\n",
+    "\n",
+    "\n",
+    "# Create pipeline instance\n",
+    "my_job = partition_job_in_pipeline(\n",
+    "    pipeline_input_data=input_oj_data,\n",
+    ")\n",
+    "\n",
+    "# Set pipeline level compute\n",
+    "my_job.tags.update\n",
```

Comment:
why need this line: 
"my_job.tags.update"

Reply:
Not sure where I copied this. Seems not useful. Will remove this from next commit.